scipy/spatial/distance.py

```diff
@@ -307,6 +311,76 @@ def _validate_weights(w, dtype=np.float64):
     return w
 
 
+def _add_periodic_bounds(fun):
+    _name = 'periodic_bounds'
+    _type = "tuple of ndarray, default: None"
+    _desc = textwrap.dedent(
+        """
+        Tuple containing the lower and upper bounds of periodic boundary conditions,
+        e.g. ``(lb, ub)``.  ``lb`` and ``ub`` must be one-dimensional arrays with the
+        same shape as arguments `u` and `v`, and all of ``ub - lb`` must be positive.
+        Elements of ``ub - lb`` that are non-finite will not be treated as periodic.
+        """
+        ).split('\n')[1:]
+    _periodic_bounds_doc = _docscrape.Parameter(_name, _type, _desc)
+    _periodic_bounds_param = inspect.Parameter(
+        _name, inspect.Parameter.KEYWORD_ONLY, default=None)
+
+    def wrap_to_bounds(x, lb, period):
+        return _lazywhere(np.isfinite(period),
+                          (x, lb, period),
+                          f=lambda x, lb, period: (x - lb) % period + lb,
+                          f2=lambda x, lb, period: x)
+
+    @functools.wraps(fun)
+    def wrapped(*args, periodic_bounds=None, **kwargs):
```

Comment:
This can probably simplified. For these metrics, I think all we need is the periodicity.